vue-app/src/utils/accounts.ts

```diff
@@ -1,5 +1,32 @@
 import { ethers } from 'ethers'
+import { mainnetProvider } from '@/api/core'
+import { isAddress } from '@ethersproject/address'
 
 export function isSameAddress(address1: string, address2: string): boolean {
   return ethers.utils.getAddress(address1) === ethers.utils.getAddress(address2)
 }
+
+export async function ensLookup(address: string): Promise<string | null> {
+  // Looks up possible ENS for given 0x address
+  const name: string | null = await mainnetProvider.lookupAddress(address)
+  return name
+}
+
+export async function resolveEns(name: string): Promise<string | null> {
+  // Returns null if the name passed is a 0x address
+  if (isAddress(name)) return null
+  // If name is valid ENS returns 0x address, else returns null
+  return await mainnetProvider.resolveName(name)
+}
+
+export async function isValidEns(name: string): Promise<boolean> {
```

Comment:
Doesn't look like we use this anywhere - should we be?

Reply:
Good catch... I was only using it in an existing approach to the form validation. This is no longer needed with `isValidEthAddress` as a replacement.